homeassistant/util/dt.py

```diff
@@ -43,7 +47,10 @@ def get_time_zone(time_zone_str: str) -> dt.tzinfo | None:
 
     Async friendly.
     """
-    return tz.gettz(time_zone_str)
+    try:
+        return zoneinfo.ZoneInfo(time_zone_str)  # type: ignore
```

Comment:
Why is `# type: ignore` necessary here? `mypy` seems to work fine for me with this on Python 3.9 (though maybe y'all are doing something weird, because the pipe syntax for returning unions is a 3.10 feature, but it seems like `tox.ini -e typing` would run on Python 3.9...

Reply:
We still support python 3.8

`homeassistant/util/dt.py:51: error: Returning Any from function declared to return "Optional[tzinfo]"  [no-any-return]`

Reply:
On Python 3.8 it considers the import from backports to be `Any`. Not sure why.